specs/scenarios/telemetry.md

```diff
@@ -0,0 +1,93 @@
+
+## Goals
+
+- Understand how Kiota is being used.
+- Understand what scenarios are important to our community.
+- Understand what are the preferred experiences for using Kiota.
+
+## Non-goals
+
+- We are not trying to collect any personally identifiable information.
+- We are not trying to collect any information about the API being used.
+- We are not trying to collect any information about the application using the API.
+- We are not trying to collect any information during the runtime of the application.
+
+## Proposal
+
+We should introduce a new telemetry component to Kiota. This component should be enabled by default and should be able to be disabled by the user, in a very similar way that the `dotnet` CLI does (https://learn.microsoft.com/en-us/dotnet/core/tools/telemetry). The telemetry component should be able to collect information about the following scenarios:
+
+- Adoption of the CLI in general.
+- Adoption of the different commands.
+- Adoption of the different parameters for each command (without values that could be considered sensitive like the OpenAPI description, include / exclude paths, etc.).
+- Adoption of the different languages.
+- Adoption of the different platforms we offer support for.
+
+### Privacy
+
+We should be very careful about the information we collect. Before rolling out this feature, we should have full agreement with CELA on our approach, the way we collect and protect the data. In general:
+
+- We should offer a way to opt-out of the telemetry collection.
+- We should not collect any information that could be considered sensitive. 
+- We should not collect any information that could be used to identify a person, an application, or an API. 
+- We should not collect any information during the runtime of the application.
+
+### Examples of questions we want to answer
+
+- How many users are using the CLI?
+- How many users are using the CLI via the extension?
+- What is the current growth trajectory of the usage of kiota?
+- What are the most used commands?
+- How are the commands used? Which parameters are being used?
+- What are the most used languages?
+- What are the most used platforms?
+- Is there a difference in how the CLI is being used between platforms?
+- Do we have users using old versions of kiota? Why?
+- When launching new capabilities, what is the adoption rate? How long does it take for users to adopt the new capabilities?
+- Can we identify a spike in error being returned by Kiota? What is the impact of the error? What command generates the error? What parameters are being used?
+
+### Data collected
+
+#### Basic data being collected
+
+For every command, we should collect the following information:
+
+- Timestamp
+- Operating system
+- Operating system version
+- Source (CLI or extension)
+- Acquisition channel (dotnet tool, binaries, homebrew, asdf, extension, etc.)
```

Comment:
I'm not sure how we're going to collect that since we don't own half of those distribution mechanisms

Reply:
I assume the exes would live in "special" folders? Couldn't we understand the CWD and map it to known channels we care / know about?

Reply:
if we deem that a best effort (those folders can generally be remapped through configuration) and assume everything is a binary (or unknown) by default, sure.

Reply:
Let's go with that for now and see how we can get it. If not, I'm open to review the spec to remove but getting that info will be interesting from a customer acquisition standpoint.